### PR TITLE
Updated the platform channels documentation for background isolate channels

### PR DESCRIPTION
**Pull request by gaaclarke**
This adds documentation for the new feature that allows us to use plugins and channels from background isolates.

issue: https://github.com/flutter/flutter/issues/13937

**Do not land until that feature is on stable.**

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

File: src/development/platform-integration/platform-channels.md [hunk 1]
```diff
@@ -969,15 +969,13 @@ types than the default types.
...
-a background thread if using a Task Queue.
-You can invoke the platform side handlers asynchronously
-and on any thread when the Task Queue API is available;
+When invoking channels on the platform side destined for Flutter, invoke them on
+the platform's main thread. When invoking channels in Flutter destined for the
+platform side, they can be invoked from any `Isolate` that is the root
+`Isolate`, or is registered as a background `Isolate`. The platform side's
+handlers can execute on the platform's main thread or they can execute on a
+background thread if using a Task Queue. You can invoke the platform side
+handlers asynchronously and on any thread when the Task Queue API is available;
```
Comment (sfshaza2) [inline src/development/platform-integration/platform-channels.md:978]: Are TaskQueues only on Android?

```suggestion
When invoking channels on the platform side destined for Flutter,
invoke them on the platform's main thread.
When invoking channels in Flutter destined for the platform side,
either invoke them from any `Isolate` that is the root
`Isolate`, _or_ that is registered as a background `Isolate`.
The handlers for the platform side can execute on the platform's main thread
or they can execute on a background thread if using a Task Queue.
You can invoke the platform side handlers asynchronously
and on any thread when the Task Queue API is available;
```

**Review by sfshaza2 — COMMENTED** [review]
lgtm once some minor suggestions are addressed

File: src/development/platform-integration/platform-channels.md (hunk 1 above)
Comment (gaaclarke) [reply src/development/platform-integration/platform-channels.md:978]: TaskQueues are only on iOS and Android.

**Comment by Leedehai** [comment]
Hi - any movements on this PR?

**Comment by stuartmorgan** [comment]
@Leedehai No, per the comment in the PR description:

> Do not land until that feature is on stable.

**Comment by atsansone** [comment]
@gaaclarke : Did this hit `stable` yet?

**Comment by gaaclarke** [comment]
> @gaaclarke : Did this hit `stable` yet?

no, siree

**Comment by sfshaza2** [comment]
@gaaclarke, I see that issue https://github.com/flutter/flutter/issues/13937 is closed because it has been fixed on master. Do you know when it will hit stable? I see a few possible approaches:

* We add a note to the text that this is fixed on master and then remove when it hits stable
* We add a note that it will be stable in release 3.3.8 (or whatever)
* We leave this PR as is until it does actually hit stable

I'm not a fan of lingering PRs, but sometimes it's just the way it is.

**Comment by gaaclarke** [comment]
> @gaaclarke, I see that issue [flutter/flutter#13937](https://github.com/flutter/flutter/issues/13937) is closed because it has been fixed on master. Do you know when it will hit stable? 

I have a rough idea.  I don't know if publicly we've promised a date for the next stable release.  It may be later than we were originally expecting.  It will be the next stable release though.

> I see a few possible approaches:
> 
> * We add a note to the text that this is fixed on master and then remove when it hits stable
> * We add a note that it will be stable in release 3.3.8 (or whatever)
> * We leave this PR as is until it does actually hit stable
> 
> I'm not a fan of lingering PRs, but sometimes it's just the way it is.

It isn't bothering me beyond the stress of making sure it doesn't get forgotten.  Any of those sounds fine to me, just let me know what works for you.  I already am in this predicament because of the medium article I'm sitting on anyways.

**Comment by sfshaza2** [comment]
ok, I'll leave this for now. But I'm happy to review your article, @gaaclarke!

**Comment by gaaclarke** [comment]
> ok, I'll leave this for now. But I'm happy to review your article, @gaaclarke!

Thanks shams, it's the one you already reviewed, not a new one =)

**Comment by sfshaza2** [comment]
closing

**Pull request closed without merging**